pkg/detector/ospkg/debian/debian.go

```diff
@@ -86,55 +85,53 @@ func (s *Scanner) Detect(osVer string, pkgs []ftypes.Package) ([]types.DetectedV
 
 	var vulns []types.DetectedVulnerability
 	for _, pkg := range pkgs {
-		// This bucket has only fixed vulnerabilities.
-		// To detect vulnerabilities, it is necessary to compare versions between the installed version and the patched version.
-		advisories, err := s.ovalVs.Get(osVer, pkg.SrcName)
-		if err != nil {
-			return nil, xerrors.Errorf("failed to get debian OVAL: %w", err)
-		}
-
 		installed := utils.FormatSrcVersion(pkg)
 		installedVersion, err := version.NewVersion(installed)
 		if err != nil {
-			log.Logger.Debugf("failed to parse Debian installed package version: %w", err)
+			log.Logger.Debugf("Debian installed package version error: %s", err)
 			continue
 		}
 
+		advisories, err := s.vs.Get(osVer, pkg.SrcName)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to get debian advisories: %w", err)
+		}
+
 		for _, adv := range advisories {
+			vuln := types.DetectedVulnerability{
+				VulnerabilityID:  adv.VulnerabilityID,
+				VendorIDs:        adv.VendorIDs,
+				PkgName:          pkg.Name,
+				InstalledVersion: installed,
+				FixedVersion:     adv.FixedVersion,
+				Layer:            pkg.Layer,
+			}
+
+			if adv.Severity != dbTypes.SeverityUnknown {
+				// Package-specific severity
+				vuln.SeveritySource = vulnerability.Debian
+				vuln.Vulnerability = dbTypes.Vulnerability{
+					Severity: adv.Severity.String(),
+				}
+			}
+
+			// It means unfixed vulnerability. We don't have to compare versions.
+			if adv.FixedVersion == "" {
```

Comment:
Won't this be always the case, as Debian buckets will only have unfixed versions?
https://github.com/aquasecurity/trivy-db/blob/726b7b72a47bff2aacd01a4ea4104c5fda3255a3/pkg/vulnsrc/debian/debian.go#L405